service/matching/matcher.go

```diff
@@ -297,13 +297,13 @@ forLoop:
 
 // Poll blocks until a task is found or context deadline is exceeded
 // On success, the returned task could be a query task or a regular task
-// Returns ErrNoTasks when context deadline is exceeded
+// Returns errNoTasks when context deadline is exceeded
```

Comment:
I think I've done the same thing before, but it's a little weird to me that we're documenting an unexported field here (which FWIW won't be linked or visible from the Godoc) from an exported method. However, since the error is only for internal usage, I think it makes sense that it's unexported, so I think we should just remove this line of documentation or otherwise remove the reference to errNoTasks.

Reply:
We should absolutely not remove this line.. it's important for callers to understand what happens when the context hits the deadline.

Whether things are exported or appear in godoc (which I doubt anyone generates or looks at for this) is irrelevant, it's just documentation on the function.